resources/lib/RequestBuilder.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+// require_once __DIR__ . '/Plugin.php';
+// use FindologicApi\Constants\Plugin;
+use FINDOLOGIC\Api\Requests\Request;
+use FINDOLOGIC\Api\Definitions\OutputAdapter;
+use FINDOLOGIC\Api\Requests\SearchNavigation\SearchNavigationRequest;
+
+class RequestBuilder extends Request
+{
+    const SORT_MAPPING = [
+        'sorting.price.avg_asc' => 'price ASC',
+        'sorting.price.avg_desc' => 'price DESC',
+        'texts.name1_asc' => 'label ASC',
+        'texts.name1_desc' => 'label DESC',
+        'variation.createdAt_desc' => 'dateadded DESC',
+        'variation.createdAt_asc' => 'dateadded ASC',
+        'variation.position_asc' => 'salesfrequency ASC',
+        'variation.position_desc' => 'salesfrequency DESC'
+    ];
+
+    private Request|SearchNavigationRequest $request;
+    private ?string $requestType = null;
+    private ?string $shopUrl = null;
+    private ?string $shopKey = null;
+    private ?string $revision = null;
+    private ?string $userIp = null;
+    private ?string $shopType = null;
+    private ?string $shopVersion = null;
+    private ?array $params = null;
+    private ?array $externalSearch = null;
+    private $isTagPage = null;
+    private $tagId = null;
+    private ?string $categoryName = null;
+    private $category = null;
+
+    public function __construct(
+        $requestType = null,
+        $shopUrl = null,
+        $shopKey = null,
+        $revision = null,
+        $userIp = null,
+        $shopType = null,
+        $shopVersion = null,
+        $params = null,
+        $externalSearch = null,
+        $isTagPage = null,
+        $tagId = null,
+        $categoryName = null,
+        $category = null
+    ) {
```

Comment:
Please use [constructor promotion](https://stitcher.io/blog/constructor-promotion-in-php-8)